### ISSUE TITLE
4.x - Route not found when using a port other than 80

### ISSUE THREAD

**Issue by dcorrea777** [slimphp/Slim#2840]
I have a slim application running in docker image with php 7.2 on my host's 8080 port.

My slim app has the following route definition

```php
<?php

use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;
use Slim\Factory\AppFactory;

require_once __DIR__ . '/../vendor/autoload.php';

$app = AppFactory::create();
$app->get('/', function (Request $request, Response $response, $args) {
    $response->getBody()->write('hello-word');
    return $response;
});
$app->get('/auth', function (Request $request, Response $response) {
    $response->getBody()->write('auth');
    return $response;
});

$app->run();
```

## Problem

When I access my application at the root address `http: // localhost: 8080` it works normally, but when I access my `auth` route, it returns me 404. (But it is set).

## Note
* This problem only occurs when I use any port other than 80, I tested on 80 and all routes worked normally.
* My mod_rwrite is enabled

I'm new to slim, I would like some help =)

**Comment by odan** [comment]
The RoutingMiddleware is essential. You can find it in the documentation on the first page:

* http://www.slimframework.com/docs/v4/
* https://odan.github.io/2019/09/09/slim-4-cheatsheet-and-faq.html#error-404

```php
$app = AppFactory::create();

// Add Routing Middleware
$app->addRoutingMiddleware();

// ...

$app->run();
```

**Comment by dcorrea777** [comment]
Hi @odan , thanks for the reply.

I added `$app->addRoutingMiddleware()`, however the problem is different.

Actually, my problem was related to my browser redirection.

Example:
```php
$app->get('auth', function (Request $request, Response $response, $args) {
// ...
});
```

In defining my route I did not put the `/` at the end every time I accessed through the browser, for some reason my browser was included with the slash (`/`) at the end, and this was causing the problem
